### PR TITLE
Add `Identity a` basic instances

### PR DESCRIPTION
**Pull request by aiya000**
I want the basic instances for `Identity a` :relaxed:

### CODE REVIEW

**Comment by Melvar** [comment]
Would you please make the `Show` instance parenthesize when necessary by implementing and using `showPrec` instead of `show`? Read the docs in `Prelude.Show`, there are helpers.

**Comment by aiya000** [comment]
@Melvar  I fixed now :+1:

I have a question, (I'm Idris beginner :joy:).

An `implementation` keyword is often shown in instance definitions,
but I think the instances can be defined without `implementation`.

What is it?
and shouldn't I use `implementation`?

**Comment by Melvar** [comment]
It needs to be `showArg x` instead of `show x`.
The implementation keyword is optional, I think in prelude and base libraries it’s usually left off.

**Comment by aiya000** [comment]
@Melvar Can I remove `implementation` in Identity.idr?
And are other things OK?
(sorry, I'm English beginner :joy:)

File: libs/base/Control/Monad/Identity.idr [hunk 1]
```diff
@@ -11,8 +11,21 @@ implementation Functor Identity where
...
 implementation Applicative Identity where
     pure x = Id x
-
     (Id f) <*> (Id g) = Id (f g)
 
 implementation Monad Identity where
     (Id x) >>= k = k x
+
+implementation Show a => Show (Identity a) where
+  showPrec d (Id x) = showCon d "Id" $ show x
```
Comment (Melvar) [inline libs/base/Control/Monad/Identity.idr]: The `show x` here needs to be `showArg x` instead.

**Comment by Melvar** [comment]
You can remove the `implementation` keyword.

**Comment by aiya000** [comment]
Thanks, I finished to add `showArg` and remove `implementation` keyword 👍

**Comment by ahmadsalim** [comment]
@aiya000 I think the idea was that you should use the `showArg` function on the argument (like you used `showCon` on the constructor), and not rename `show` to `showArg` 😄

**Comment by aiya000** [comment]
@ahmadsalim Thank you! I might understand 😄 
@Melvar The task maybe is done, would you please check it :D

**Comment by Melvar** [comment]
Looks correct now.

**Comment by ahmadsalim** [comment]
Cool, thanks!

**Pull request merged**